### PR TITLE
Rename Twig extensions according to best practices

### LINKED ISSUE #176: Refactor Twig extension names according to best practices guild
According to the Symfony2 best practices guild (http://symfony.com/doc/current/best_practices/templates.html#twig-extensions) twig extension naming convension would be the bundle name followed by underscore (_) and the extension name

ex: app_locale_extension (not app.locale_extension)

### PR DESCRIPTION
**Pull request by bocharsky-bw**
This fixes #176.

### CODE REVIEW

File: src/AppBundle/Twig/AppExtension.php [hunk 1]
```diff
@@ -52,6 +52,6 @@ public function markdownToHtml($content)
     // using 'app.extension' if you only have one Twig extension in your application.
```
Comment (sebastianblum) [inline src/AppBundle/Twig/AppExtension.php]: you should change it also here in the comment

Comment (bocharsky-bw) [inline src/AppBundle/Twig/AppExtension.php]: Oh, thank you! I definitely missed it.

**Comment by voronkovich** [comment]
I think the `_extension` suffix (we don't use a suffix `_type` in form types) not needed. Also I think we should rename `AppExtension` to `MarkdownExtension`.

**Comment by bocharsky-bw** [comment]
Well, `app_extension` came from [best practices](http://symfony.com/doc/current/best_practices/templates.html#twig-extensions) like a `_extension` suffix. In this case we should change it in best practices first.

File: src/AppBundle/Twig/AppExtension.php [hunk 2]
```diff
@@ -49,9 +49,9 @@ public function markdownToHtml($content)
     }
 
     // the name of the Twig extension must be unique in the application. Consider
-    // using 'app.extension' if you only have one Twig extension in your application.
+    // using 'app_extension' if you only have one Twig extension in your application.
```
Comment (voronkovich) [inline src/AppBundle/Twig/AppExtension.php]: @bocharsky-bw,  see this comment about one extension in the appplication.

Comment (bocharsky-bw) [inline src/AppBundle/Twig/AppExtension.php]: What if we had `UserType` form type and `UserExtension` twig extension in app bundle? Then we had a collision with `app_user` if we remove `_extension` suffix. So I don't like idea to remove `_extension` suffix. However, form types use much often then Twig extensions, so they had exclusion from `_type` suffix. I'll be better to add `_type` suffix to form type than remove `_extension` one.

Comment (stof) [inline src/AppBundle/Twig/AppExtension.php]: form type names and twig extension names cannot conflict. They are identifiers in different systems. We will never register a form type in Twig or a twig extension in the form registry (it is impossible)

Comment (bocharsky-bw) [inline src/AppBundle/Twig/AppExtension.php]: @stof So we could get rid of `_extension` suffix?

Comment (stof) [inline src/AppBundle/Twig/AppExtension.php]: yes

File: app/config/services.yml [hunk 3]
```diff
@@ -11,7 +11,7 @@ services:
     # using them in the templates
     app.twig.app_extension:
```
Comment (xabbuh) [inline app/config/services.yml]: Imo the service name should be changed too.

Comment (bocharsky-bw) [inline app/config/services.yml]: Oh, thanks! I missed it

**Comment by xabbuh** [comment]
:+1:

File: src/AppBundle/Twig/MarkdownExtension.php [hunk 4]
```diff
@@ -24,7 +24,7 @@
  * @author Ryan Weaver <weaverryan@gmail.com>
  * @author Javier Eguiluz <javier.eguiluz@gmail.com>
  */
-class AppExtension extends \Twig_Extension
```
Comment (stof) [inline src/AppBundle/Twig/MarkdownExtension.php]: there is a comment referring to this file in SourceCodeExtension. Please update it too

Comment (bocharsky-bw) [inline src/AppBundle/Twig/MarkdownExtension.php]: Thank you. I updated

**Comment by javiereguiluz** [comment]
My only concern with this change is the following:

1) Previously we had an `AppExtension`, where you can obviously define all the custom filters and functions your application needs (we onle had one, but we could add more in the future).
2) Now we have a `MarkdownExtension` which obviously is just for Markdown and nothing else.

Some developers may think that you must define a Twig extension for each filter/function.

**Comment by bocharsky-bw** [comment]
You're right, @javiereguiluz . So maybe better to refactor all Twig extensions with `AppExtension` except `SourceCodeExtension`?

**Comment by stof** [comment]
the solution could indeed be to put the locale stuff in AppExtension, showing a case where the different functions needed by the app are provided by a single extension (SourceCodeExtension should indeed stay separate as it is not really here to be part of the application, but to expose a debugging stuff for the demo)

**Comment by voronkovich** [comment]
I disagree with @javiereguiluz. I think that `AppExtension` is only good for importing native php functions (and simple custom functions) into Twig templates. We shouldn't put code that have no cohesion in a one class.

**Comment by bocharsky-bw** [comment]
I think Javier idea is nice if you have only a few Twig filters like in our case.

**Comment by bocharsky-bw** [comment]
So what to do? Refactor all Twig extensions except `SourceCodeExtension` with `AppExtension`?

**Comment by javiereguiluz** [comment]
@bocharsky-bw I'd like to do exactly that, but let's wait a bit to see if anyone proposes something different. Thanks.

**Comment by javiereguiluz** [comment]
Closing it in favor of #220, which implements the idea of merging all Twig extensions.

**Pull request closed without merging**